training/training/core/celery/worker.py

```diff
@@ -0,0 +1,194 @@
+from datetime import date
+from celery import Celery
+import django
+from django.conf import settings
+
+settings.configure()
+django.setup()
+
+import boto3
+
+
+from training.core.celery.criterion import getCriterionHandler
+from training.core.celery.dataset import SklearnDatasetCreator
+from training.core.celery.dataset import ImageDefaultDatasetCreator
+from training.core.celery.dl_model import DLModel
+from torch.utils.data import DataLoader
+from training.core.celery.optimizer import getOptimizer
+from training.core.celery.trainer import ClassificationTrainer, RegressionTrainer
+from training.routes.tabular.schemas import TabularParams
+from training.routes.image.schemas import ImageParams
+from training.routes.training.results.schemas import (
+    DetailedTrainResultsData,
+    TrainResultsData,
+)
+
+from training import celeryconfig
+
+celery_app = Celery("celery")
+celery_app.config_from_object(celeryconfig)
+
+
+def saveDetailedTrainResultsDataToS3(
+    detailedTrainResultsData: DetailedTrainResultsData,
+):
+    s3 = boto3.resource("s3")
+    s3.Object(
+        "dlp-executions", f"{detailedTrainResultsData.basic_info.trainspaceId}.json"
```

Comment:
question: at the point where celery executes the training request, do we already know which user id a given trainspace id belongs to?

Reply:
yes